src/commands/findRoom.ts

```diff
@@ -0,0 +1,477 @@
+		const error: GetRoomsResponse = {
+			Rooms: [],
+		};
+		return error;
+	}
+}
+async function _getRoomsBetween(
+	search_type: string,
+	building: string,
+	timeA: string,
+	timeB: string
+): Promise<GetRoomsResponse> {
+	try {
+		const { data, status } = await axios.get<GetRoomsResponse>(
+			`https://pi.zyancey.com/${search_type}/${building}/${timeA}/${timeB}`
+		);
+		if (status !== 200) throw new Error(`${status}`);
+		return data;
+	} catch (error_) {
+		logger.error('Error in getting Room Info:');
+		logger.error(error_);
+		const error: GetRoomsResponse = {
+			Rooms: [],
+		};
+		return error;
+	}
+}
+
+async function _getWhenRoom(
+	search_type: string,
+	building: string,
+	room: string
+): Promise<GetRoomInfoResponse> {
+	try {
+		const { data, status } = await axios.get<GetRoomInfoResponse>(
+			`https://pi.zyancey.com/${search_type}/${building}/${room}`
+		);
+		if (status !== 200) throw new Error(`${status}`);
+		return data;
+	} catch (error_) {
+		logger.error('Error in getting Room Info:');
+		logger.error(error_);
+		const error: GetRoomInfoResponse = {
+			busySince: 'Error',
+			busyUntil: 'Error',
+			isInUse: false,
+		};
+		return error;
+	}
+}
+
+const builder = new SlashCommandBuilder()
+	.setName('findroom')
+	.setDescription('Finds you a room to study in on campus!')
+	.addSubcommand(subcommand =>
+		subcommand
+			.setName('now')
+			.setDescription('Finds you an available room on campus right now!')
+			.addStringOption(option =>
+				option
+					.setName('building')
+					.setDescription('The building acronym to search in, type any to see all options')
+					.addChoices(
+						{ name: 'ANY', value: 'ANY' },
+						{ name: 'BNSN', value: 'BNSN' },
+						{ name: 'BRMB', value: 'BRMB' },
+						{ name: 'CTB', value: 'CTB' },
+						{ name: 'EB', value: 'EB' },
+						{ name: 'ELLB', value: 'ELLB' },
+						{ name: 'ESC', value: 'ESC' },
+						{ name: 'HBLL', value: 'HBLL' },
+						{ name: 'JFSB', value: 'JFSB' },
+						{ name: 'JKB', value: 'JKB' },
+						{ name: 'JRCB', value: 'JRCB' },
+						{ name: 'JSB', value: 'JSB' },
+						{ name: 'KMBL', value: 'KMBL' },
+						{ name: 'LSB', value: 'LSB' },
+						{ name: 'MARB', value: 'MARB' },
+						{ name: 'MB', value: 'MB' },
+						{ name: 'MCDB', value: 'MCDB' },
+						{ name: 'MCKB', value: 'MCKB' },
+						{ name: 'MSRB', value: 'MSRB' },
+						{ name: 'SLC', value: 'SLC' },
+						{ name: 'TLRB', value: 'TLRB' },
+						{ name: 'TMCB', value: 'TMCB' },
+						{ name: 'TNRB', value: 'TNRB' },
+						{ name: 'WSC', value: 'WSC' },
+						{ name: 'WVB', value: 'WVB' }
+					)
+					.setRequired(true)
+			)
+	)
+	.addSubcommand(subcommand =>
+		subcommand
+			.setName('at')
+			.setDescription('Finds you an available room on campus at a given time!')
+			.addStringOption(option =>
+				option
+					.setName('start_time')
+					.setDescription('What time would you like to search for?')
+					.addChoices(
+						{ name: '8:00 AM', value: '08:00:00' },
```

Comment:
Might be wise to break this list of times into an array, then re-use that array in the places it's needed